android/engine/src/main/java/org/smartregister/fhircore/engine/task/WorkflowCarePlanGenerator.kt

```diff
@@ -131,33 +139,18 @@ constructor(
     data: Bundle = Bundle(),
     output: CarePlan,
   ) {
-    val patientId = IdType(patient.id).idPart
-
     if (cqlLibraryIdList.isEmpty()) {
       loadPlanDefinitionResourcesFromDb()
     }
 
-    val r4PlanDefinitionProcessor = createPlanDefinitionProcessor()
-    // TODO Fix after resolving dependency issues
-    /*val carePlanProposal = CarePlan()
-    r4PlanDefinitionProcessor.
-     apply(
-       id =  null,
-       canonical =  planDefinition.idElement,
-       planDefinition =  null,
-       subject =  patientId,
-       encounterId =  null,
-       practitionerId =  null,
-       organizationId =  null,
-       userType =  null,
-       userLanguage =  null,
-       userTaskContext =  null,
-       setting =  null,
-       settingContext =  null
-    ) as CarePlan*/
+    val carePlanProposal =
+      fhirOperator.generateCarePlan(
+        planDefinition,
+        patient,
+        data,
+      ) as CarePlan
 
-    // Accept the proposed (transient) CarePlan by default and add tasks to the CarePlan of record
-    //    acceptCarePlan(carePlanProposal, output)
+    acceptCarePlan(carePlanProposal, output)
 
     resolveDynamicValues(
       planDefinition = planDefinition,
```

Comment:
This should not be needed since the Apply operation does resolve the dynamic values. However, many of the test cases have dynamic values that will not be executed by the default operation. A custom implementation might be needed if those values are necessary.